### PR TITLE
[Doc] [Clusters] Add missing steps and simplify YAML in top-level clusters quickstart

### LINKED ISSUE #35299: [docs][clusters] Getting Started for Clusters is missing steps
### Description

The Cluster Quickstart in the top-level Getting Started needs to be fixed because it:
* is missing steps.
* has a cluster config that is way too complicated.

See https://docs.ray.io/en/master/cluster/vms/getting-started.html

cc: @matthewdeng 

### Link

https://docs.ray.io/en/master/ray-overview/getting-started.html

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Closes https://github.com/ray-project/ray/issues/35299 by adding missing steps and using a minimal cluster YAML file.

Addresses https://github.com/ray-project/ray/issues/33026 for the AWS example-minimal.yaml by adding min_workers, max_workers and the instance type.  The issue will remain open until it's fixed for all the other cluster provider example-minimal YAML files as well.

I'll run through the final tutorial manually once before merging.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 1]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
...
+        min_workers: 1
+        # The maximum number of worker nodes of this type to launch.
+        # This takes precedence over min_workers.
+        max_workers: 4
+        # Provider-specific config for this node type, e.g. instance type. By default
+        # Ray will auto-configure unspecified fields such as SubnetId and KeyName.
+        # For more documentation on available fields, see:
+        # http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+        node_config:
+            InstanceType: m5.large
```
Comment (rynewang) [inline python/ray/autoscaler/aws/example-minimal.yaml:32]: newline?

**Review by rynewang — APPROVED** [review]

**Review by jjyao — APPROVED** [review]

File: doc/source/ray-overview/getting-started.md [hunk 2]
```diff
@@ -530,12 +530,24 @@ Learn more about Ray Core
...
 Ray programs can run on a single machine, or seamlessly scale to large clusters.
+
+:::{note}
+To run this example install the following:
+
+```bash
+pip install -U "ray[default]" boto3
+```
+
+If you haven't already, you’ll need to configure your credentials as described in the https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#guide-credentials[documentation for boto3].
```
Comment (angelinalg) [inline doc/source/ray-overview/getting-started.md:548]:
```suggestion
If you haven't already, configure your credentials as described in the https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#guide-credentials[documentation for boto3].
```

File: doc/source/ray-overview/getting-started.md [hunk 3]
```diff
@@ -548,13 +560,13 @@ Take this simple example that waits for individual nodes to join the cluster.
 You can also download this example from our [GitHub repository](https://github.com/ray-project/ray/blob/master/doc/yarn/example.py).
```
Comment (angelinalg) [inline doc/source/ray-overview/getting-started.md:560]:
```suggestion
You can also download this example from the [GitHub repository](https://github.com/ray-project/ray/blob/master/doc/yarn/example.py).
```

File: doc/source/ray-overview/getting-started.md [hunk 4]
```diff
@@ -548,13 +560,13 @@ Take this simple example that waits for individual nodes to join the cluster.
...
 Go ahead and store it locally in a file called `example.py`.
```
Comment (angelinalg) [inline doc/source/ray-overview/getting-started.md:561]:
```suggestion
Store it locally in a file called `example.py`.
```

File: doc/source/ray-overview/getting-started.md [hunk 5]
```diff
@@ -548,13 +560,13 @@ Take this simple example that waits for individual nodes to join the cluster.
...
 
-To execute this script in the cloud, just download [this configuration file](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml),
+To execute this script in the cloud, just download [this configuration file](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-minimal.yaml),
```
Comment (angelinalg) [inline doc/source/ray-overview/getting-started.md:563]:
```suggestion
To execute this script in the cloud, download [this configuration file](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-minimal.yaml),
```

File: doc/source/ray-overview/getting-started.md [hunk 6]
```diff
@@ -570,7 +582,7 @@ ray submit cluster.yaml example.py --start
 :outline:
 :expand:
 
-Learn more about launching Ray Clusters
+Learn more about launching Ray Clusters on AWS, GCP, Azure and more
```
Comment (angelinalg) [inline doc/source/ray-overview/getting-started.md:585]:
```suggestion
Learn more about launching Ray Clusters on AWS, GCP, Azure, and more
```

File: doc/source/ray-overview/getting-started.md [hunk 7]
```diff
@@ -530,12 +530,24 @@ Learn more about Ray Core
 
 ## Ray Cluster Quickstart
 
-Deploy your applications on Ray clusters, often with minimal code changes to your existing code.
+Deploy your applications on Ray clusters on AWS, GCP, Azure and more, often with minimal code changes to your existing code.
```
Comment (angelinalg) [inline doc/source/ray-overview/getting-started.md:533]:
```suggestion
Deploy your applications on Ray clusters on AWS, GCP, Azure, and more, often with minimal code changes to your existing code.
```

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 8]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
 provider:
     type: aws
     region: us-west-2
+
+# Tell the autoscaler the allowed node types and the resources they provide.
+# The key is the name of the node type, which is just for debugging purposes.
```
Comment (angelinalg) [inline python/ray/autoscaler/aws/example-minimal.yaml:10]:
```suggestion
# The key is the name of the node type, which is for debugging purposes.
```

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 9]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
...
+# The node config specifies the launch config and physical instance type.
+available_node_types:
+    ray.head.default:
+        # Provider-specific config for this node type, e.g. instance type. By default
```
Comment (angelinalg) [inline python/ray/autoscaler/aws/example-minimal.yaml:14]:
```suggestion
        # Provider-specific config for this node type, e.g., instance type. By default
```

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 10]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
...
+        # Ray will auto-configure unspecified fields such as SubnetId and KeyName.
```
Comment (angelinalg) [inline python/ray/autoscaler/aws/example-minimal.yaml:15]:
```suggestion
        # Ray auto-configures unspecified fields such as SubnetId and KeyName.
```

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 11]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
...
+        # For more documentation on available fields, see:
```
Comment (angelinalg) [inline python/ray/autoscaler/aws/example-minimal.yaml:16]:
```suggestion
        # For more documentation on available fields, see:
```
```suggestion
        # For more documentation on available fields, see
```

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 12]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
...
+        # http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+        node_config:
+            InstanceType: m5.large
+    ray.worker.default:
+        # The minimum number of worker nodes of this type to launch.
+        # This number should be >= 0.
+        min_workers: 1
+        # The maximum number of worker nodes of this type to launch.
+        # This takes precedence over min_workers.
```
Comment (angelinalg) [inline python/ray/autoscaler/aws/example-minimal.yaml:25]:
```suggestion
        # This parameter takes precedence over min_workers.
```

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 13]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
...
+        max_workers: 4
+        # Provider-specific config for this node type, e.g. instance type. By default
```
Comment (angelinalg) [inline python/ray/autoscaler/aws/example-minimal.yaml:27]:
```suggestion
        # Provider-specific config for this node type, e.g., instance type. By default
```

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 14]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
...
+        # Ray will auto-configure unspecified fields such as SubnetId and KeyName.
```
Comment (angelinalg) [inline python/ray/autoscaler/aws/example-minimal.yaml:28]:
```suggestion
        # Ray auto-configures unspecified fields such as SubnetId and KeyName.
```

File: python/ray/autoscaler/aws/example-minimal.yaml [hunk 15]
```diff
@@ -5,3 +5,28 @@ cluster_name: aws-example-minimal
...
+        # For more documentation on available fields, see:
```
Comment (angelinalg) [inline python/ray/autoscaler/aws/example-minimal.yaml:29]:
```suggestion
        # For more documentation on available fields, see
```

**Review by angelinalg — APPROVED** [review]

**Comment by architkulkarni** [comment]
The script passed: 

```
[...]
Counter({('ip-10-0-218-191', 'ip-10-0-228-193'): 8, ('ip-10-0-225-188', 'ip-10-0-225-188'): 8, ('ip-10-0-225-188', 'ip-10-0-218-191'): 8, ('ip-10-0-218-191', 'ip-10-0-218-191'): 8, ('ip-10-0-225-188', 'ip-10-0-215-149'): 8, ('ip-10-0-218-191', 'ip-10-0-225-188'): 7, ('ip-10-0-225-188', 'ip-10-0-228-193'): 7, ('ip-10-0-215-149', 'ip-10-0-218-191'): 7, ('ip-10-0-228-193', 'ip-10-0-225-188'): 7, ('ip-10-0-215-149', 'ip-10-0-215-149'): 6, ('ip-10-0-215-149', 'ip-10-0-228-193'): 6, ('ip-10-0-218-191', 'ip-10-0-215-149'): 6, ('ip-10-0-215-149', 'ip-10-0-225-188'): 5, ('ip-10-0-228-193', 'ip-10-0-215-149'): 4, ('ip-10-0-228-193', 'ip-10-0-228-193'): 3, ('ip-10-0-228-193', 'ip-10-0-218-191'): 2})
Success!
Shared connection to 44.229.14.251 closed.
```

**Pull request merged**